benchmarks/src/main/java/org/opensearch/benchmark/store/remote/filecache/FileCacheBenchmark.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.benchmark.store.remote.filecache;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.lucene.store.IndexInput;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.opensearch.index.store.remote.filecache.CachedIndexInput;
+import org.opensearch.index.store.remote.filecache.FileCache;
+import org.opensearch.index.store.remote.filecache.FileCacheFactory;
+
+/**
+ * Simple benchmark test of {@link FileCache}. It uses a uniform random distribution
+ * of keys, which is very simple but unlikely to be representative of any real life
+ * workload.
+ */
+@Warmup(iterations = 1)
+@Measurement(iterations = 1)
+@Fork(1)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+@Threads(8)
+@SuppressWarnings("unused") // invoked by benchmarking framework
+public class FileCacheBenchmark {
+    private static final CachedIndexInput INDEX_INPUT = new FixedSizeStubIndexInput();
+
+    @Benchmark
+    public void get(CacheParameters parameters) {
+        parameters.fileCache.get(randomKeyInCache(parameters));
```

Comment:
Curious if the benchmark should use blackhole or return the value, the call could be eliminated since the returned value is not used, thoughts?

Reply:
Yeah, good call.

My guess is the call would not be eliminated because the JVM doesn't know if there are no side effects to invoking the `get()` method. I'm certainly no expert here but as a general practice I think the return values should be blackholed.

Reply:
Right, that should make it:

```
    public Object get(CacheParameters parameters) {
        return parameters.fileCache.get(randomKeyInCache(parameters));
    }
```

Reply:
I went with:

```
@Benchmark
public void get(CacheParameters parameters, Blackhole blackhole) {
    blackhole.consume(parameters.fileCache.get(randomKeyInCache(parameters)));
}
```

Is that not the right way to use the blackhole here?

Reply:
My apologies @andrross,  probably the comment went into outdated code version, both approaches are correct, thank you.